From f5f5d5322fb74dd793f1fb0d95a97c9667c2a0ed Mon Sep 17 00:00:00 2001
From: honjow <honjow311@gmail.com>
Date: Tue, 3 Mar 2026 20:45:10 +0800
Subject: [PATCH] PCI/PM: Add MSI MS-1T41 DMI quirk to skip forced D0 init

---
 drivers/pci/pci.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/pci/pci.c b/drivers/pci/pci.c
index 9e42090fb1089..8838c8c41b5fb 100644
--- a/drivers/pci/pci.c
+++ b/drivers/pci/pci.c
@@ -3198,6 +3198,17 @@ void pci_pm_power_up_and_verify_state(struct pci_dev *pci_dev)
 	pci_update_current_state(pci_dev, PCI_D0);
 }
 
+static const struct dmi_system_id pci_pm_skip_d0_init[] = {
+	{
+		/* MSI Claw variants on MS-1T41 board */
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Micro-Star International Co., Ltd."),
+			DMI_MATCH(DMI_BOARD_NAME, "MS-1T41"),
+		},
+	},
+	{ }
+};
+
 /**
  * pci_pm_init - Initialize PM functions of given PCI device
  * @dev: PCI device to handle.
@@ -3270,7 +3281,8 @@ void pci_pm_init(struct pci_dev *dev)
 	if (status & PCI_STATUS_IMM_READY)
 		dev->imm_ready = 1;
 poweron:
-	pci_pm_power_up_and_verify_state(dev);
+	if (!dmi_check_system(pci_pm_skip_d0_init))
+		pci_pm_power_up_and_verify_state(dev);
 	pm_runtime_forbid(&dev->dev);
 	pm_runtime_set_active(&dev->dev);
 	pm_runtime_enable(&dev->dev);
-- 
2.51.2

